File: space-game/Core.Tests/Extensions/EnumerableExtensionsTests.vb
Imports Core.Classes
Imports Core.Enums
Imports Core.Extensions
Imports Core.Tests.Data
Imports NUnit.Framework

Namespace Extensions
    <TestFixture>
    Public Class EnumerableExtensionsTests
        <TestCaseSource(GetType(EnumerableExtensionsTestsData), NameOf(EnumerableExtensionsTestsData.HasAny_WhenGivenAList_ReturnsExpected_Data))>
        Public Sub HasAny_WhenGivenAList_ReturnsExpected(list As List(Of String), expected As Boolean)
            Dim result = list.HasAny()

            Assert.AreEqual(expected, result)
        End Sub

        <TestCaseSource(GetType(EnumerableExtensionsTestsData), NameOf(EnumerableExtensionsTestsData.GetHighestAvailableResolution_WhenGivenAList_ReturnsExpected_Data))>
        Public Sub GetHighestAvailableResolution_WhenGivenAList_ReturnsExpected(textures As List(Of Texture), type As TextureType, expected As String)
            Dim result = textures.GetHighestAvailableResolution(type)

            Assert.AreEqual(expected, result)
        End Sub

        <TestCaseSource(GetType(EnumerableExtensionsTestsData), NameOf(EnumerableExtensionsTestsData.GetLowestAvailableResolution_WhenGivenAList_ReturnsExpected_Data))>
        Public Sub GetLowestAvailableResolution_WhenGivenAList_ReturnsExpected(textures As List(Of Texture), type As TextureType, expected As String)
            Dim result = textures.GetLowestAvailableResolution(type)

            Assert.AreEqual(expected, result)
        End Sub

        <TestCaseSource(GetType(EnumerableExtensionsTestsData), NameOf(EnumerableExtensionsTestsData.GetHighestAvailableResolutionForEachType_High))>
        <TestCaseSource(GetType(EnumerableExtensionsTestsData), NameOf(EnumerableExtensionsTestsData.GetHighestAvailableResolutionForEachType_Medium))>
        <TestCaseSource(GetType(EnumerableExtensionsTestsData), NameOf(EnumerableExtensionsTestsData.GetHighestAvailableResolutionForEachType_Low))>
        <TestCaseSource(GetType(EnumerableExtensionsTestsData), NameOf(EnumerableExtensionsTestsData.GetHighestAvailableResolutionForEachType_Mix))>
        <TestCaseSource(GetType(EnumerableExtensionsTestsData), NameOf(EnumerableExtensionsTestsData.GetHighestAvailableResolutionForEachType_Empty))>
        Public Sub GetHighestAvailableResolutionForEachType_WhenGivenAList_ReturnsExpected(textures As List(Of Texture), expected As List(Of Texture))
            Dim result = textures.GetHighestAvailableResolutionForEachType()

            Assert.AreEqual(expected, result)
        End Sub

        <TestCaseSource(GetType(EnumerableExtensionsTestsData), NameOf(EnumerableExtensionsTestsData.IsEquivalent_WhenGivenAList_ReturnsExpected_NullAndEmpty))>
        <TestCaseSource(GetType(EnumerableExtensionsTestsData), NameOf(EnumerableExtensionsTestsData.IsEquivalent_WhenGivenAList_ReturnsExpected_Contents))>
        Public Sub IsEquivalent_WhenGivenAList_ReturnsExpected(one As List(Of Texture), two As List(Of Texture), expected As Boolean)
            Dim result = one.IsEquivalent(two)

            Assert.AreEqual(expected, result)
        End Sub
    End Class
End Namespace

File: space-game/Core.Tests/Extensions/EnumerableExtensionsTests.cs
using Core.Classes;
using Core.Enums;
using Core.Extensions;
using Core.Tests.Data;
using NUnit.Framework;
using System.Collections.Generic;

namespace Core.Tests.Extensions
{
    [TestFixture]
    public class EnumerableExtensionsTests
    {
        [TestCaseSource(typeof(EnumerableExtensionsTestsData), nameof(EnumerableExtensionsTestsData.HasAny_WhenGivenAList_ReturnsExpected_Data))]
        public void HasAny_WhenGivenAList_ReturnsExpected(List<string> list, bool expected)
        {
            bool result = list.HasAny();

            Assert.AreEqual(expected, result);
        }

        [TestCaseSource(typeof(EnumerableExtensionsTestsData), nameof(EnumerableExtensionsTestsData.GetHighestAvailableResolution_WhenGivenAList_ReturnsExpected_Data))]
        public void GetHighestAvailableResolution_WhenGivenAList_ReturnsExpected(List<Texture> textures, TextureType type, string expected)
        {
            string result = textures.GetHighestAvailableResolution(type);

            Assert.AreEqual(expected, result);
        }

        [TestCaseSource(typeof(EnumerableExtensionsTestsData), nameof(EnumerableExtensionsTestsData.GetLowestAvailableResolution_WhenGivenAList_ReturnsExpected_Data))]
        public void GetLowestAvailableResolution_WhenGivenAList_ReturnsExpected(List<Texture> textures, TextureType type, string expected)
        {
            string result = textures.GetLowestAvailableResolution(type);

            Assert.AreEqual(expected, result);
        }

        [TestCaseSource(typeof(EnumerableExtensionsTestsData), nameof(EnumerableExtensionsTestsData.GetHighestAvailableResolutionForEachType_High))]
        [TestCaseSource(typeof(EnumerableExtensionsTestsData), nameof(EnumerableExtensionsTestsData.GetHighestAvailableResolutionForEachType_Medium))]
        [TestCaseSource(typeof(EnumerableExtensionsTestsData), nameof(EnumerableExtensionsTestsData.GetHighestAvailableResolutionForEachType_Low))]
        [TestCaseSource(typeof(EnumerableExtensionsTestsData), nameof(EnumerableExtensionsTestsData.GetHighestAvailableResolutionForEachType_Mix))]
        [TestCaseSource(typeof(EnumerableExtensionsTestsData), nameof(EnumerableExtensionsTestsData.GetHighestAvailableResolutionForEachType_Empty))]
        public void GetHighestAvailableResolutionForEachType_WhenGivenAList_ReturnsExpected(List<Texture> textures, List<Texture> expected)
        {
            List<Texture> result = textures.GetHighestAvailableResolutionForEachType();

            Assert.AreEqual(expected, result);
        }

        [TestCaseSource(typeof(EnumerableExtensionsTestsData), nameof(EnumerableExtensionsTestsData.IsEquivalent_WhenGivenAList_ReturnsExpected_NullAndEmpty))]
        [TestCaseSource(typeof(EnumerableExtensionsTestsData), nameof(EnumerableExtensionsTestsData.IsEquivalent_WhenGivenAList_ReturnsExpected_Contents))]
        public void IsEquivalent_WhenGivenAList_ReturnsExpected(List<Texture> one, List<Texture> two, bool expected)
        {
            bool result = one.IsEquivalent(two);

            Assert.AreEqual(expected, result);
        }
    }
}
